### ISSUE TITLE
Ignoring changes to cluster_encryption_config using very old version 4.0.2

### ISSUE THREAD

**Issue by nodesocket** [terraform-aws-modules/terraform-aws-eks#1266]
We are running a very old version of this module `4.0.2` but because we are using Terraform `0.11.13` we really can't upgrade .

I enabled KMS secret encryption on a KMS cluster manually via the AWS console, but now Terraform wants to disable it because it's not defined in the Terraform module. Unfortunately `4.0.2` does not have the concept of `cluster_encryption_config` and I can't define it.

Making matters even worse, I can't `ignore_changes` on `cluster_encryption_config` because that is not supported in modules in Terraform version `0.11.X`.

So, essentially I am stuck now unable to apply any Terraform changes because KMS secret encryption cannot be disabled once enabled and I can't update my Terraform version.

Any ideas?

```
encryption_config.#:                                                                                   "1" => "0" (forces new resource)
encryption_config.0.provider.#:                                                                        "1" => "0" (forces new resource)
encryption_config.0.provider.0.key_arn:                                                                "arn:aws:kms:us-east-2:051898693016:key/06a70608-ebc2-4677-9e84-99d282bdcd1b" => "" (forces new resource)
encryption_config.0.resources.#:                                                                       "1" => "0" (forces new resource)
encryption_config.0.resources.2031673519:                                                              "secrets" => "" (forces new resource)
 ```

**Comment by nodesocket** [comment]
I was able to **hack** around this by adding:

```
lifecycle {
    ignore_changes = [
      "encryption_config",
    ]
  }
```

Inside of the module filesin `.terraform/modules/<hash>/cluster.tf` on line 24. Not a great solution, but wondering if there is a better way.

**Comment by schollii** [comment]
You really need to set aside time to upgrade to terraform 0.14. It is really worth the effort. 

And actually it is not that much effort, the 2 upgrade commands included in tf 0.12 and 0.13  will handle most of the tedium, then for 0.14 you may have to fix some provider blocks. 

The longer you wait, the bigger the problem will get, because you will have dependencies that have to be upgraded too and some with backwards incompatible changes, so then the upgrade process requires to fix issues, rename resources in your tf state file, etc.
